specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json

```diff
@@ -3589,6 +3589,60 @@
         },
         "x-ms-long-running-operation": true
       },
+      "patch": {
+        "tags": [
+          "VirtualMachineScaleSetExtensions"
+        ],
+        "operationId": "VirtualMachineScaleSetExtensions_Update",
+        "description": "The operation to update an extension.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmScaleSetName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the VM scale set where the extension should be updated."
+          },
+          {
+            "name": "vmssExtensionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the VM scale set extension."
+          },
+          {
+            "name": "extensionParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/VirtualMachineScaleSetExtensionUpdate"
```

Comment:
do you need to have "VirtualMachineScaleSetExtensionUpdate" definition?  or can we just use "VirtualMachineScaleSetExtension"

Reply:
For the Patch api, the response http status code should be only 200 (OK). While for Put, it can be 200(OK) or 201(Created)